File: bot-api/src/main/java/gg/mineral/bot/api/math/BoundingBox.java
package gg.mineral.bot.api.math;

public interface BoundingBox {

    /**
     * Sets the minimum x value of the bounding box.
     * 
     * @param minX
     *             the minimum x value of the bounding box
     */
    void setMinX(double minX);

    /**
     * Sets the minimum y value of the bounding box.
     * 
     * @param minY
     *             the minimum y value of the bounding box
     */
    void setMinY(double minY);

    /**
     * Sets the minimum z value of the bounding box.
     * 
     * @param minZ
     *             the minimum z value of the bounding box
     */
    void setMinZ(double minZ);

    /**
     * Sets the maximum x value of the bounding box.
     * 
     * @param maxX
     *             the maximum x value of the bounding box
     */
    void setMaxX(double maxX);

    /**
     * Sets the maximum y value of the bounding box.
     * 
     * @param maxY
     *             the maximum y value of the bounding box
     */
    void setMaxY(double maxY);

    /**
     * Sets the maximum z value of the bounding box.
     * 
     * @param maxZ
     *             the maximum z value of the bounding box
     */
    void setMaxZ(double maxZ);

    /**
     * Gets the minimum x value of the bounding box.
     * 
     * @return the minimum x value of the bounding box
     */
    double getMinX();

    /**
     * Gets the minimum y value of the bounding box.
     * 
     * @return the minimum y value of the bounding box
     */
    double getMinY();

    /**
     * Gets the minimum z value of the bounding box.
     * 
     * @return the minimum z value of the bounding box
     */
    double getMinZ();

    /**
     * Gets the maximum x value of the bounding box.
     * 
     * @return the maximum x value of the bounding box
     */
    double getMaxX();

    /**
     * Gets the maximum y value of the bounding box.
     * 
     * @return the maximum y value of the bounding box
     */
    double getMaxY();

    /**
     * Gets the maximum z value of the bounding box.
     * 
     * @return the maximum z value of the bounding box
     */
    double getMaxZ();

    /**
     * Checks if the specified vector is inside the bounding box.
     * 
     * @param x
     *          the x-coordinate of the vector
     * @param y
     *          the y-coordinate of the vector
     * @param z
     *          the z-coordinate of the vector
     * @return {@code true} if the vector is inside the bounding box, otherwise
     *         {@code false}
     */
    default boolean isVecInside(double x, double y, double z) {
        return x >= getMinX() && x < getMaxX() && y >= getMinY() && y < getMaxY() && z >= getMinZ() && z < getMaxZ();
    }
}

File: bot-api/src/main/kotlin/gg/mineral/bot/api/math/BoundingBox.kt
package gg.mineral.bot.api.math

interface BoundingBox {
    /**
     * Gets the minimum x value of the bounding box.
     *
     * @return the minimum x value of the bounding box
     */
    /**
     * Sets the minimum x value of the bounding box.
     *
     * @param minX
     * the minimum x value of the bounding box
     */
    var minX: Double

    /**
     * Gets the minimum y value of the bounding box.
     *
     * @return the minimum y value of the bounding box
     */
    /**
     * Sets the minimum y value of the bounding box.
     *
     * @param minY
     * the minimum y value of the bounding box
     */
    var minY: Double

    /**
     * Gets the minimum z value of the bounding box.
     *
     * @return the minimum z value of the bounding box
     */
    /**
     * Sets the minimum z value of the bounding box.
     *
     * @param minZ
     * the minimum z value of the bounding box
     */
    var minZ: Double

    /**
     * Gets the maximum x value of the bounding box.
     *
     * @return the maximum x value of the bounding box
     */
    /**
     * Sets the maximum x value of the bounding box.
     *
     * @param maxX
     * the maximum x value of the bounding box
     */
    var maxX: Double

    /**
     * Gets the maximum y value of the bounding box.
     *
     * @return the maximum y value of the bounding box
     */
    /**
     * Sets the maximum y value of the bounding box.
     *
     * @param maxY
     * the maximum y value of the bounding box
     */
    var maxY: Double

    /**
     * Gets the maximum z value of the bounding box.
     *
     * @return the maximum z value of the bounding box
     */
    /**
     * Sets the maximum z value of the bounding box.
     *
     * @param maxZ
     * the maximum z value of the bounding box
     */
    var maxZ: Double

    /**
     * Checks if the specified vector is inside the bounding box.
     *
     * @param x
     * the x-coordinate of the vector
     * @param y
     * the y-coordinate of the vector
     * @param z
     * the z-coordinate of the vector
     * @return `true` if the vector is inside the bounding box, otherwise
     * `false`
     */
    fun isVecInside(x: Double, y: Double, z: Double): Boolean {
        return x >= minX && x < maxX && y >= minY && y < maxY && z >= minZ && z < maxZ
    }
}
